src/core/control/SearchControl.cpp

```diff
@@ -9,39 +9,26 @@
 #include "model/Text.h"     // for Text
 #include "model/XojPage.h"  // for XojPage
 #include "view/TextView.h"  // for TextView
+#include "view/overlays/SearchResultView.h"
 
-using std::string;
+SearchControl::SearchControl(const PageRef& page, XojPdfPageSPtr pdf):
+        page(page),
+        pdf(std::move(pdf)),
+        viewPool(std::make_shared<xoj::util::DispatchPool<xoj::view::SearchResultView>>()) {}
 
-SearchControl::SearchControl(const PageRef& page, XojPdfPageSPtr pdf) {
-    this->page = page;
-    this->pdf = std::move(pdf);
-}
-
-SearchControl::~SearchControl() { freeSearchResults(); }
-
-void SearchControl::freeSearchResults() { this->results.clear(); }
-
-void SearchControl::paint(cairo_t* cr, double zoom, const GdkRGBA& color) {
-    // set the line always the same size on display
-    cairo_set_line_width(cr, 1 / zoom);
-
-    for (XojPdfRectangle rect: this->results) {
-        cairo_rectangle(cr, rect.x1, rect.y1, rect.x2 - rect.x1, rect.y2 - rect.y1);
-        gdk_cairo_set_source_rgba(cr, &color);
-        cairo_stroke_preserve(cr);
-        auto applied = GdkRGBA{color.red, color.green, color.blue, 0.3};
-        gdk_cairo_set_source_rgba(cr, &applied);
-        cairo_fill(cr);
-    }
-}
-
-auto SearchControl::search(std::string text, int* occures, double* top) -> bool {
-    freeSearchResults();
+SearchControl::~SearchControl() { this->results.clear(); }
```

Comment:
nit: won't `results` will be cleared automatically when it's destroyed?
```suggestion
SearchControl::~SearchControl() = default;
```